__tests__/app.test.js

```diff
@@ -1,46 +1,83 @@
+/** * @jest-environment node */
 const request = require('supertest');
-const app = require('../src/app');
 const nock = require('nock');
+const app = require('../src/app');
+const { mockListOfJokes, mockRandomJokes, mockPersonalJokes } = require('../data/test-data');
 
-describe('My jokes API', () => {
-  describe('API GET /', () => {
-    it('should respond with a welcome message', done => {
-      request(app)
-        .get('/')
-        .then(res => {
-          expect(res.statusCode).toEqual(200);
-          expect(res.body.message).toEqual('Welcome to my jokes API!');
-          done();
-        });
-    });
-  });
-  it('that this is the jokes endpoint', done => {
+describe('GET / - Homepage', () => {
+  it('should respond with some homepage markup', async () => {
     request(app)
-      .get('/jokes')
+      .get('/')
       .then(res => {
         expect(res.statusCode).toEqual(200);
-        expect(res.body.jokes).toEqual(
-          'You have reached the jokes endpoint, please laugh with care',
-        );
-        done();
+        expect(res.text).toContain('Hello, Welcome to my jokes API');
       });
   });
-  it('should respond with a random joke message', done => {
-    request(app)
+});
+
+describe('GET /jokes', () => {
+  it('that jokes endpoint responds with list of jokes', async () => {
+    nock('https://api.icndb.com')
+      .get('/jokes')
+      .reply(200, mockListOfJokes);
+
+    const res = await request(app).get('/jokes');
+    expect(res.statusCode).toEqual(200);
+    expect(res.body.jokes).toEqual(mockListOfJokes.value);
+  });
+  it('should respond with an error message if something goes wrong', async () => {
+    nock('https://api.icndb.com')
+      .get('/jokes')
+      .replyWithError({ statusCode: 500, message: 'huge error' });
+
+    const res = await request(app).get('/jokes');
+    expect(res.statusCode).toEqual(500);
+    expect(res.body.error).toEqual('huge error');
+  });
+});
+
+describe('GET, /jokes/random', () => {
+  it('should respond with a random joke', async () => {
+    nock('https://api.icndb.com')
       .get('/jokes/random')
-      .then(res => {
-        expect(res.statusCode).toEqual(200);
-        expect(res.body.message).toEqual('No jokes here yet, try again!');
-        done();
-      });
+      .query({ exclude: '[explicit]' })
+      .reply(200, mockRandomJokes);
+
+    const res = await request(app).get('/jokes/random');
+    expect(res.statusCode).toEqual(200);
+    expect(res.body.randomJoke).toEqual(mockRandomJokes.value);
   });
-  it('test to get jokes with personalised names ', done => {
-    request(app)
-      .get('/jokes/random/personal/Saint/Last')
-      .then(res => {
-        expect(res.statusCode).toEqual(200);
-        expect(res.body.message).toEqual('Hi, i can personalise your jokes!');
-        done();
-      });
+  it('should respond with an error message if something goes wrong', async () => {
```

Comment:
You need to be more sepcific with your test name here. You are only testing for a 404, which does not cover when something goes wrong every time. You could get a 400, a 500 etc